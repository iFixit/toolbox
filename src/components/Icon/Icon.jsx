import React from 'react';
import PropTypes from 'prop-types';
import glamorous from 'glamorous';
import { toSvg } from 'feather-icons';

const SvgContainer = glamorous.span(
   {
      display: 'inline-block',
   },
   ({ size, color, strokeWidth }) => ({
      width: size,
      height: size,
      color,
      strokeWidth,
   }),
);

const Icon = props =>
   <SvgContainer
      className={props.className}
      size={props.size}
      color={props.color}
      strokeWidth={props.strokeWidth}
      dangerouslySetInnerHTML={{
         __html: toSvg(props.name, {
            width: '100%',
            height: '100%',
            'stroke-width': 'inherit',
         }),
      }}
   />;

Icon.propTypes = {
   /** Icon name. See the complete list of icons at https://feathericons.com */
   name: PropTypes.string.isRequired,
   /** Set class name of containing element. */
   className: PropTypes.string,
   /** Any legal CSS color value for the stroke color */
   color: PropTypes.string,
   /** Width and height of the icon in pixels */
   size: PropTypes.number,
   /** Stroke width of the icon in pixels */
   strokeWidth: PropTypes.number,
};

Icon.defaultProps = {
   className: '',
   color: 'currentColor',
   size: 24,
   strokeWidth: 2,
};

export default Icon;
