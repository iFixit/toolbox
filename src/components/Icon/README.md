```
<Icon name="home" />
```

```
<Icon name="thumbs-up" size={48} strokeWidth={1} color="#0071CE" />
```