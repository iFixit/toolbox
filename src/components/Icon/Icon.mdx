---
name: Icon
route: /components/icon
menu: Components
---

import Icon from './Icon'
import color from '../../constants/color/color'
import { Playground, PropsTable } from 'docz'

# Icon

## Examples

<Playground>
  <Icon name="home" />
</Playground>

<Playground>
  <Icon name="thumbs-up" size={48} strokeWidth={1} color={color.blue[4]} />
</Playground>

## Props

<PropsTable of={Icon} />
