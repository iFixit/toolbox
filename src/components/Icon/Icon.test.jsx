import React from 'react';
import renderer from 'react-test-renderer';
import Icon from './Icon';

test('renders without crashing', () => {
   const tree = renderer.create(<Icon name="square" />).toJSON();
   expect(tree).toMatchSnapshot();
});

test('renders with custom color, size and stroke width', () => {
   const tree = renderer
      .create(<Icon name="home" color="#f00" size={16} strokeWidth={1} />)
      .toJSON();
   expect(tree).toMatchSnapshot();
});
