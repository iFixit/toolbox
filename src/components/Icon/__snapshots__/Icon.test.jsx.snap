// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with custom color, size and stroke width 1`] = `
.glamor-0,
[data-glamor-0] {
  display: inline-block;
  width: 16px;
  height: 16px;
  color: #f00;
  stroke-width: 1;
}

<span
  className="glamor-0"
  dangerouslySetInnerHTML={
    Object {
      "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"100%\\" height=\\"100%\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" stroke=\\"currentColor\\" stroke-width=\\"inherit\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" class=\\"feather feather-home\\"><path d=\\"M3 9l9-7 9 7v11a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2z\\"></path><polyline points=\\"9 22 9 12 15 12 15 22\\"></polyline></svg>",
    }
  }
/>
`;

exports[`renders without crashing 1`] = `
.glamor-0,
[data-glamor-0] {
  display: inline-block;
  width: 24px;
  height: 24px;
  color: currentColor;
  stroke-width: 2;
}

<span
  className="glamor-0"
  dangerouslySetInnerHTML={
    Object {
      "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"100%\\" height=\\"100%\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" stroke=\\"currentColor\\" stroke-width=\\"inherit\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" class=\\"feather feather-square\\"><rect x=\\"3\\" y=\\"3\\" width=\\"18\\" height=\\"18\\" rx=\\"2\\" ry=\\"2\\"></rect></svg>",
    }
  }
/>
`;
